packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -0,0 +1,255 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:pub_semver/pub_semver.dart';
+
+import '../base/common.dart';
+import '../base/context.dart';
+import '../base/file_system.dart';
+import '../base/os.dart';
+import '../base/platform.dart';
+import '../base/process_manager.dart';
+import '../globals.dart';
+import '../ios/plist_utils.dart';
+
+AndroidStudio get androidStudio =>
+    context.putIfAbsent(AndroidStudio, AndroidStudio.latestValid);
+
+// Android Studio layout:
+
+// Linux/Windows:
+// $HOME/.AndroidStudioX.Y/system/.home
+
+// macOS:
+// /Applications/Android Studio.app/Contents/
+// $HOME/Applications/Android Studio.app/Contents/
+
+// $STUDIO_HOME/gradle/gradle-X.Y.Z/bin/gradle
+
+final Version minGradleVersion = new Version(2, 14, 1);
+
+/// Locate Gradle.
+String get gradleExecutable {
+  // See if the user has explicitly configured gradle-dir.
+  String gradleDir = config.getValue('gradle-dir');
+  if (gradleDir != null) {
+    if (fs.isFileSync(gradleDir)) return gradleDir;
```

Comment:
Which doesn't really make sense, given that you also have the 'no braces for single-line if statements' rule. Together, those make the code less readable, at least to me.

But seriously, isn't it time to ditch the Special Snowflake style rules, and just embrace dartfmt? Makes life so much simpler! ;)